src/riak_kv_index_hashtree.erl

```diff
@@ -1029,3 +1039,8 @@ maybe_get_vnode_lock(SrcPartition, Pid) ->
         false ->
             ok
     end.
+
+maybe_callback(undefined) ->
+    ok;
+maybe_callback(Callback) ->
+    Callback().
```

Comment:
Yep, I thought Fred switched it for the reasoning you mentioned Doug... as it was a catch in the first commit. 

Zeeshan Lakhani
programmer | 
software engineer at @basho | 
org. member/founder of @papers_we_love | paperswelove.org
twitter => @zeeshanlakhani

> On Mar 7, 2016, at 9:29 AM, Doug Rohrer <notifications@github.com> wrote:
> 
> In src/riak_kv_index_hashtree.erl:
> 
> > @@ -1029,3 +1039,8 @@ maybe_get_vnode_lock(SrcPartition, Pid) ->
> >          false ->
> >              ok
> >      end.
> > +
> > +maybe_callback(undefined) ->
> > +    ok;
> > +maybe_callback(Callback) ->
> > +    Callback().
> This should probably be catch Callback(). unless we're expecting the consumer to be able to cancel the update by throwing (which would seem more appropriate to do with a return value from Callback() anyway).
> 
> —
> Reply to this email directly or view it on GitHub.
>